<launch>
    <!--  XBee Nodes-->
    <node pkg="xbee_localization" type="base_station.py" name="xbee_communication" args="/dev/ttyUSB0 9600" output="screen"/>
    <node pkg="xbee_localization" type="remote_request.py" name="beacon_2" args='beacon_2 0 2' output="screen"/>
    <node pkg="xbee_localization" type="remote_request.py" name="beacon_1" args='beacon_1 0 1' output="screen"/>
    <node pkg="xbee_localization" type="remote_request.py" name="beacon_3" args='beacon_3 0 3' output="screen"/>

    <node pkg="xbee_localization" type="message_demux.py" name="demux" output="screen"/>
    <node pkg="tf" type="static_transform_publisher" name="beacon_1_tf" args="-0.46 4.65 0 0 0 0 /map /beacon_1 100"/>
    <node pkg="tf" type="static_transform_publisher" name="beacon_2_tf" args="-0.46 -2 0 0 0 0 /map /beacon_2 100"/>
    <node pkg="tf" type="static_transform_publisher" name="beacon_3_tf" args="2 -9 0 0 0 0 /map /beacon_3 100"/>
    <!-- Location of the WiFi Beacon -->
    <node pkg="tf" type="static_transform_publisher" name="wifi_tf" args="-0.4 0 0 0 0 0 /map /wifi 100"/>

    <!-- WiFi Observations -->
    <node pkg="wireless_watcher" type="watcher_node" name="watcher_node" output="screen"/>
    <node pkg="rosbag" type="record" name="rosbag" args="/xbee_request /xbee_response /xbee_response/beacon1 /xbee_response/beacon2 /xbee_response/beacon3 /connected /connection /tf /map /tf_static /map_metadata /map_updates /graft/state -o /tmp/indoor_localization"/>
</launch>
